File: tests/fixtures/brownfield/go_stdlib_legacy_quote/quote/service.go
package quote

import (
	"encoding/json"
	"errors"
	"fmt"
	"net/http"
	"strings"
)

func QuoteOrder(unitPriceCents int, quantity int) (int, error) {
	if unitPriceCents < 0 {
		return 0, errors.New("unit price must not be negative")
	}
	if quantity <= 0 {
		return 0, errors.New("quantity must be positive")
	}
	return unitPriceCents * quantity, nil
}

func FormatReceipt(totalCents int) string {
	return fmt.Sprintf("Total: %d.%02d", totalCents/100, totalCents%100)
}

func NormalizeCoupon(code string) string {
	return strings.ToUpper(strings.TrimSpace(code))
}

func LegacyDiscountHint(code string) *int {
	if NormalizeCoupon(code) != "SAVE5" {
		return nil
	}
	value := 5
	return &value
}

type quoteOrderBody struct {
	UnitPriceCents *int `json:"unit_price_cents"`
	Quantity       *int `json:"quantity"`
}

func Handler() http.Handler {
	mux := http.NewServeMux()
	mux.HandleFunc("POST /quote-order", func(response http.ResponseWriter, request *http.Request) {
		decoder := json.NewDecoder(request.Body)
		decoder.DisallowUnknownFields()
		var body quoteOrderBody
		if err := decoder.Decode(&body); err != nil {
			writeJSON(response, http.StatusBadRequest, map[string]any{"error": "invalid request"})
			return
		}
		if body.UnitPriceCents == nil {
			writeJSON(response, http.StatusBadRequest, map[string]any{"error": "unit price must be an integer"})
			return
		}
		if body.Quantity == nil {
			writeJSON(response, http.StatusBadRequest, map[string]any{"error": "quantity must be an integer"})
			return
		}
		total, err := QuoteOrder(*body.UnitPriceCents, *body.Quantity)
		if err != nil {
			writeJSON(response, http.StatusBadRequest, map[string]any{"error": err.Error()})
			return
		}
		writeJSON(response, http.StatusOK, map[string]any{
			"receipt":     FormatReceipt(total),
			"total_cents": total,
		})
	})
	return mux
}

func writeJSON(response http.ResponseWriter, status int, value any) {
	response.Header().Set("content-type", "application/json")
	response.WriteHeader(status)
	if err := json.NewEncoder(response).Encode(value); err != nil {
		panic(err)
	}
}

File: tests/fixtures/brownfield/typescript_fastify_legacy_quote/src/service.ts
import Fastify, { type FastifyInstance } from "fastify";

type QuoteOrderBody = {
  readonly unit_price_cents?: unknown;
  readonly quantity?: unknown;
};

function requireInteger(value: unknown, field: string): number {
  if (typeof value !== "number" || !Number.isInteger(value)) {
    throw new Error(`${field} must be an integer`);
  }
  return value;
}

export function quoteOrder(
  unitPriceCents: number,
  quantity: number,
): number {
  if (unitPriceCents < 0) {
    throw new Error("unit price must not be negative");
  }
  if (quantity <= 0) {
    throw new Error("quantity must be positive");
  }
  return unitPriceCents * quantity;
}

export function formatReceipt(totalCents: number): string {
  return `Total: ${(totalCents / 100).toFixed(2)}`;
}

export function normalizeCoupon(code: string): string {
  return code.trim().toUpperCase();
}

export function legacyDiscountHint(code: string): number | null {
  return normalizeCoupon(code) === "SAVE5" ? 5 : null;
}

export function buildApp(): FastifyInstance {
  const app = Fastify({ logger: false });

  app.post<{ Body: QuoteOrderBody }>("/quote-order", async (request, reply) => {
    try {
      const unitPriceCents = requireInteger(
        request.body.unit_price_cents,
        "unit price",
      );
      const quantity = requireInteger(request.body.quantity, "quantity");
      const totalCents = quoteOrder(unitPriceCents, quantity);
      return reply.status(200).send({
        total_cents: totalCents,
        receipt: formatReceipt(totalCents),
      });
    } catch (error: unknown) {
      const message = error instanceof Error ? error.message : "invalid request";
      return reply.status(400).send({ error: message });
    }
  });

  return app;
}
